### PR TITLE
ReactOnRails version 10

### PR DESCRIPTION
**Pull request by Judahmeek**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/951)

### CODE REVIEW

**Comment by justin808** [comment]
LGTM so far.

1. Doc changes?
2. Changes to spec/dummy and other tests to use the new API?

---

Reviewed 25 of 25 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[app/helpers/react_on_rails_helper.rb, line 324 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/951#-KvkyvGtn0o3YtoH98V9:-KvkyvGtn0o3YtoH98VA:bl4d2w0) ([raw file](https://github.com/shakacode/react_on_rails/blob/e8fcaf27ba8a858f492a31396b07338a4ed95623/app/helpers/react_on_rails_helper.rb#L324)):*
> ```Ruby
>   end
> 
>   def internal_react_component(component_name, raw_options = {})
> ```

should this be inside "private"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/951#-:-KvkzGPSBAOJhH-qOjz8:b9uxqrt)*

**Comment by justin808** [comment]
:lgtm:

---

Reviewed 19 of 19 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/951#-:-KwdbcbIe-s4Oip6SWrv:bnfp4nl)*

File: package.json [hunk 1]
```diff
@@ -87,7 +88,7 @@
   },
   "homepage": "https://github.com/shakacode/react_on_rails#readme",
   "dependencies": {
-    "react-on-rails": "^9.0.0-beta.12",
+    "react-on-rails": "^9.0.3",
```
Comment (shepmaster) [inline package.json:91]: Why does React on Rails version 10 require React on Rails version 9?

Comment (justin808) [reply package.json:91]: @Judahmeek This should be fixed.

Comment (justin808) [reply package.json:91]: @shepmaster Until we release, we can't update the version in the PR.

Comment (shepmaster) [reply package.json:91]: > Until we release 

I noticed it because the 10.0 release of this package requires the 9.0 release of this package 😸

Comment (tricknotes) [reply package.json:91]: It seems [react-on-rails@10.0.0](https://www.npmjs.com/package/react-on-rails) is released.
I send a PR to update this version. https://github.com/shakacode/react_on_rails/pull/959

**Pull request merged**